cis194/week2/zhansongl/LogAnalysis.hs

```diff
@@ -0,0 +1,63 @@
+module LogAnalysis where
+
+import Log
+
+import Text.Megaparsec hiding (parse)
+import Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer as L
+import Data.Void
+
+type Parser = Parsec Void String
+
+messageTypeParser :: Parser MessageType
+messageTypeParser = do
+  charType <- oneOf "IEW"
+  case charType of
+       'I' -> return Info
+       'W' -> return Warning
+       'E' -> do 
+          space1
+          errorCode <- L.decimal
+          return (Error errorCode)
+
+messageParser :: Parser LogMessage
+messageParser = do
+  messageType <- messageTypeParser
+  space1
+  timeStamp <- L.decimal
+  space
+  message <- many asciiChar
+  return $ LogMessage messageType timeStamp message
```

Comment:
I think `messageParser` can be the below. 
```Haskell
LogMessage 
<$> (messageTypeParser <* space1) 
<*> (L.decimal <* space) 
<*> (many asciiChar)
```

Reply:
will fix